File: trunk/openrest4j-api/src/com/googlecode/openrest/Status.java
package com.googlecode.openrest;

import java.io.Serializable;

import org.codehaus.jackson.annotate.JsonIgnoreProperties;
import org.codehaus.jackson.map.annotate.JsonSerialize;

@JsonIgnoreProperties(ignoreUnknown = true)
public class Status implements Serializable {
    /** Available. */
    public static final String STATUS_AVAILABLE = "available";
    /** Unavailable. */
    public static final String STATUS_UNAVAILABLE = "unavailable";

    public Status(String status, java.util.Date until) {
        this.status = status;
        this.until = ((until != null) ? until.getTime() : null);
    }

    /** Default constructor for JSON deserialization. */
    public Status() {}

    public java.util.Date until() {
        return ((until != null) ? new java.util.Date(until) : null);
    }

    public boolean available() {
        return STATUS_AVAILABLE.equals(status);
    }

    @JsonSerialize(include = JsonSerialize.Inclusion.NON_NULL)
    public String status;

    @JsonSerialize(include = JsonSerialize.Inclusion.NON_NULL)
    public Long until;

    private static final long serialVersionUID = 1L;
}

File: trunk/openrest4net-api/com/googlecode/openrest/Status.cs
using System;
using System.Collections.Generic;

namespace com.googlecode.openrest
{
    class Status
    {
        /** Available. */
        public const string STATUS_AVAILABLE = "available";
        /** Unavailable. */
        public const string STATUS_UNAVAILABLE = "unavailable";

        public Status(string status, long? until)
        {
            this.status = status;
            this.until = until;
        }

        /** Empty constructor required for initialization from JSON-encoded string. */
        internal Status() { }

        public string status;
        public long? until;
    }
}
